eclair-core/src/main/scala/fr/acinq/eclair/package.scala

```diff
@@ -185,4 +209,60 @@ package object eclair {
     // @formatter:on
   }
 
+  // returns all watch addresses with label IMPORTED
+  def listImported(bitcoinClient: ExtendedBitcoinClient)(implicit ec: ExecutionContext): Future[List[WatchAddressItem]] = {
+    bitcoinClient.listReceivedByAddress(minConf = 1, includeEmpty = true, includeWatchOnly = true).map(_.filter(_.label == "IMPORTED"))
+  }
+
+  // extracts rescan info from each channel data
+  def getRescanInfo(channel: HasCommitments) = channel match {
+    case DATA_NORMAL(_, shortChannelId, _, _, _, _, _)             => Some(Left(shortChannelId))
+    case DATA_WAIT_FOR_FUNDING_CONFIRMED(_, _, waitingSince, _, _) => Some(Right(waitingSince))
+    case DATA_WAIT_FOR_FUNDING_LOCKED(_, shortChannelId, _)        => Some(Left(shortChannelId))
+    case DATA_SHUTDOWN(_, _, _)                                    => None
```

Comment:
When they are imported `importmulti` won't consider any timestamp for their rescan, in some scenarios this might lead to missing a `WatchSpent` for the funding (especially for DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT).

Reply:
For all the four states mentioned the risk is that we miss the `WatchSpent` of the funding transaction, that's because without information on the rescan height/timestamp we can't properly instruct bitcoind on how to track their address. The import routine calls `importmulti` with all the missing addresses and their timestamps resulting in the blocks being scanned from the earliest creation time, this might coincidentally rescan funding transaction for the aforementioned states but of course it's no silver bullet and can't work reliably. Another option to import those states would be to assign them an arbitrary creation time, i.e ~1 year before the current system time.